bundles/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/GroupRenderer.java

```diff
@@ -10,42 +10,57 @@
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.emf.common.util.EList;
 import org.eclipse.smarthome.model.sitemap.Group;
+import org.eclipse.smarthome.model.sitemap.LinkableWidget;
 import org.eclipse.smarthome.model.sitemap.Widget;
 import org.eclipse.smarthome.ui.classic.render.RenderException;
 import org.eclipse.smarthome.ui.classic.render.WidgetRenderer;
 
 /**
- * This is an implementation of the {@link WidgetRenderer} interface, which
- * can produce HTML code for Group widgets.
+ * This is an implementation of the {@link WidgetRenderer} interface, which can
+ * produce HTML code for Group widgets.
  *
  * @author Kai Kreuzer - Initial contribution and API
+ * @author Gaël L'hopital - Added expended behaviour handling
  *
  */
 public class GroupRenderer extends AbstractWidgetRenderer {
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean canRender(Widget w) {
-        return w instanceof Group;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public EList<Widget> renderWidget(Widget w, StringBuilder sb) throws RenderException {
-        String snippet = getSnippet("group");
-
-        snippet = StringUtils.replace(snippet, "%id%", itemUIRegistry.getWidgetId(w));
-        snippet = StringUtils.replace(snippet, "%icon%", escapeURLPath(itemUIRegistry.getIcon(w)));
-        snippet = StringUtils.replace(snippet, "%label%", getLabel(w));
-
-        // Process the color tags
-        snippet = processColor(w, snippet);
-
-        sb.append(snippet);
-        return null;
-    }
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
```

Comment:
your PR contains pseudo changes, which seem to be due to the fact that you introduces tabs instead of spaces. Could you please make sure to use the code formatter?